dhcpv4/dhcpv4.go

```diff
@@ -214,11 +214,26 @@ func RequestFromOffer(offer DHCPv4) (*DHCPv4, error) {
 	d.AddOption(&OptMessageType{MessageType: MessageTypeRequest})
 	d.AddOption(&OptRequestedIPAddress{RequestedAddr: offer.YourIPAddr()})
 	d.AddOption(&OptServerIdentifier{ServerID: serverIP})
-	// the End option has to be added explicitly
-	d.AddOption(&OptionGeneric{OptionCode: OptionEnd})
 	return d, nil
 }
 
+// NewReply builds a DHCPv4 reply from a request.
+func NewReply(request *DHCPv4) (*DHCPv4, error) {
```

Comment:
shall we use the usual naming `NewReplyFromSomething` ?

Reply:
Ok, does `NewReplyFromRequest` sounds good?